src/main/java/com/aws/greengrass/mqtt/bridge/clients/MQTTClient.java

```diff
@@ -281,17 +281,25 @@ private synchronized void doConnect() throws MqttException, KeyStoreException {
     private void reconnectAndResubscribe() {
         int waitBeforeRetry = MIN_WAIT_RETRY_IN_SECONDS;
 
-        while (!mqttClientInternal.isConnected()) {
+        while (!mqttClientInternal.isConnected() && !Thread.currentThread().isInterrupted()) {
             try {
                 // TODO: Clean up this loop
                 doConnect();
             } catch (MqttException | KeyStoreException e) {
+                if (e.getCause() instanceof InterruptedException) {
+                    // paho doesn't reset the interrupt flag
+                    Thread.currentThread().interrupt();
+                    LOGGER.atDebug().setCause(e.getCause()).log("Interrupted during reconnect");
```

Comment:
not reason to log the exception here, we know it is an interrupt.

Reply:
👍  removed